sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -176,6 +177,195 @@ case class Like(left: Expression, right: Expression, escapeChar: Char)
   }
 }
 
+abstract class LikeAllBase extends Expression with ImplicitCastInputTypes with NullIntolerant {
+  def value: Expression = children.head
+  def list: Seq[Expression] = children.tail
+  def isNot: Boolean
+
+  override def inputTypes: Seq[AbstractDataType] = {
+    val arrayOrStr = TypeCollection(ArrayType(StringType), StringType)
+    StringType +: Seq.fill(children.size - 1)(arrayOrStr)
+  }
+
+  override def dataType: DataType = BooleanType
+
+  override def foldable: Boolean = value.foldable && list.forall(_.foldable)
+
+  override def nullable: Boolean = true
+
+  def escape(v: String): String = StringUtils.escapeLikeRegex(v, '\\')
+
+  def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).matches()
+
+  override def eval(input: InternalRow): Any = {
+    val evaluatedValue = value.eval(input)
+    if (evaluatedValue == null) {
+      null
+    } else {
+      list.foreach { e =>
+        val str = e.eval(input)
+        if (str == null) {
+          return null
+        }
+        val regex = Pattern.compile(escape(str.asInstanceOf[UTF8String].toString))
+        if(regex == null) {
```

Comment:
can this happen?

Reply:
SELECT company FROM like_all_table WHERE company LIKE ALL ('%oo%', null);